shared/prow/prow_test.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// prow_test.go contains unit tests for prow package
+
+package prow
+
+import (
+	"testing"
+	"os"
+)
+
+const(
+	repoName = "test-repo"
+	invalidJobType = "invalid"
+)
+
+func TestGetArtifacts(t *testing.T) {
+	dir := os.Getenv("ARTIFACTS")
+
+	// Test we can read from the env var
+	os.Setenv("ARTIFACTS", "test")
+	v := GetLocalArtifactsDir()
+	if v != "test" {
+		t.Fatalf("Actual artifacts dir: '%s' and Expected: 'test'", v)
+	}
+
+	// Test we can use the default
+	os.Setenv("ARTIFACTS", "")
+	v = GetLocalArtifactsDir()
+	if v != "artifacts" {
+		t.Fatalf("Actual artifacts dir: '%s' and Expected: 'artifacts'", v)
+	}
+
+	// Set it to the originial value
+	os.Setenv("ARTIFACTS", dir)
+}
+
+func TestJobPath(t *testing.T) {
```

Comment:
You didn't use a data-driven approach here. Instead, you just combined 3 tests, which didn't add any value. In go they're called "table driven tests", when you have a common test code that iterates over test data to cover different variations of a test scenario. See https://github.com/golang/go/wiki/TableDrivenTests

Reply:
good to learn, changed to use "table driven tests"